### PR TITLE
Add dummy endpoints for kms integration

### PR DESCRIPTION
**Pull request by reivaj05**
Add dummy endpoints for kms integration

Implementation will come in another PR

### CODE REVIEW

File: swagger-console.yml [hunk 1]
```diff
@@ -2867,6 +2867,315 @@ paths:
...
+          description: A successful response.
+          schema:
+            $ref: "#/definitions/kmsStatusResponse"
+        default:
+          description: Generic error response.
+          schema:
+            $ref: "#/definitions/error"
+      tags:
+        - KMS
+  /kms/key/create:
```
Comment (Alevsk) [inline swagger-console.yml:2885]: Why don't have just a `/kms/key` endpoint? to perform `create`, `delete` and `get` operations via `POST`, `GET` and `DELETE` ?

To list all the keys we can have `GET /kms/keys`
To import a key we can have `POST /kms/import/key`

File: swagger-console.yml [hunk 2]
```diff
@@ -2867,6 +2867,315 @@ paths:
...
+          description: A successful response.
+          schema:
+            $ref: "#/definitions/kmsListKeysResponse"
+        default:
+          description: Generic error response.
+          schema:
+            $ref: "#/definitions/error"
+      tags:
+        - KMS
+  /kms/key/status:
```
Comment (Alevsk) [inline swagger-console.yml:2944]:
```suggestion
  /kms/key/{id}:
```
Passing the key id should return information about the key including the status

File: swagger-console.yml [hunk 3]
```diff
@@ -2867,6 +2867,315 @@ paths:
...
+    get:
+      summary: KMS key status
+      operationId: KMSKeyStatus
+      parameters:
+        - name: keyID
+          description: KMS key id
+          in: query
+          required: true
+          type: string
```
Comment (Alevsk) [inline swagger-console.yml:2953]:
```suggestion
        - name: id
          description: KMS key id
          in: path
          required: true
          type: string
```

File: swagger-console.yml [hunk 4]
```diff
@@ -2867,6 +2867,315 @@ paths:
...
+      responses:
+        201:
+          description: A successful response.
+        default:
+          description: Generic error response.
+          schema:
+            $ref: "#/definitions/error"
+      tags:
+        - KMS
+  /kms/key/import:
```
Comment (Alevsk) [inline swagger-console.yml:2904]:
```suggestion
  /kms/import/key:
```
we can reverse this if we want to support the `/kms/key/{id}` endpoint

File: swagger-console.yml [hunk 5]
```diff
@@ -2867,6 +2867,315 @@ paths:
...
+      responses:
+        201:
+          description: A successful response.
+        default:
+          description: Generic error response.
+          schema:
+            $ref: "#/definitions/error"
+      tags:
+        - KMS
+  /kms/key/list:
```
Comment (Alevsk) [inline swagger-console.yml:2923]:
```suggestion
  /kms/keys:
```
we can rename this if we want to support the `/kms/key/{id}` endpoint

File: swagger-console.yml [hunk 6]
```diff
@@ -2867,6 +2867,315 @@ paths:
...
+          description: A successful response.
+          schema:
+            $ref: "#/definitions/kmsKeyStatusResponse"
+        default:
+          description: Generic error response.
+          schema:
+            $ref: "#/definitions/error"
+      tags:
+        - KMS
+  /kms/key/delete:
```
Comment (Alevsk) [inline swagger-console.yml:2965]:
```suggestion
  /kms/key/{id}:
```

We can merge this endpoint with the one above and use the `DELETE` method

File: swagger-console.yml [hunk 7]
```diff
@@ -2867,6 +2867,315 @@ paths:
...
+    delete:
+      summary: KMS delete key
+      operationId: KMSDeleteKey
+      parameters:
+        - name: keyID
+          description: KMS key id
+          in: query
+          required: true
+          type: string
```
Comment (Alevsk) [inline swagger-console.yml:2974]:
```suggestion
        - name: id
          description: KMS key id
          in: path
          required: true
          type: string
```

**Review by Alevsk — DISMISSED** [review]

File: swagger-console.yml (hunk 1 above)
Comment (reivaj05) [reply swagger-console.yml:2885]: Not sure, i just followed the same convention   as endpoints are in kes, that way all repos have the same endpoints. Is It ok they aré diferente? 🧐

File: restapi/admin_kms.go [hunk 8]
```diff
@@ -0,0 +1,260 @@
+// This file is part of MinIO Console Server
+// Copyright (c) 2021 MinIO, Inc.
```
Comment (jinapurapu) [inline restapi/admin_kms.go:2]:
```suggestion
// Copyright (c) 2022 MinIO, Inc.
```

**Review by jinapurapu — CHANGES_REQUESTED** [review]

**Review by bexsoft — APPROVED** [review]

**Review by jinapurapu — DISMISSED** [review]

**Review by bexsoft — APPROVED** [review]
LGTM

**Review by dvaldivia — APPROVED** [review]

**Pull request merged**